### PR TITLE
HPCC-16976 Fix external binary leak when asyncBackup is off

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: dali/base/dasds.cpp [hunk 1]
```diff
@@ -1217,6 +1217,7 @@ class CBackupHandler : public CInterface, implements IThreaded
         if (!async)
         {
             writeExt(fname, length, data);
+            free(data);
             return;
```
Comment (mckellyln) [inline dali/base/dasds.cpp]: Because its detached() right ?
Looks fine.

Comment (mckellyln) [inline dali/base/dasds.cpp]: Should you add a if (data) check ?

**Comment by mckellyln** [comment]
Minor comments, looks good.

File: dali/base/dasds.cpp (hunk 1 above)
Comment (jakesmith) [inline dali/base/dasds.cpp]: It's not possible to be null.

**Comment by jakesmith** [comment]
@richardkchapman - please merge.

**Pull request merged**